app/scripts/metamask-controller.js

```diff
@@ -1986,36 +1986,28 @@ export default class MetamaskController extends EventEmitter {
   }
 
   /**
-   * A method for providing our API over a stream using Dnode.
+   * A method for providing our API over a stream using JSON-RPC.
    * @param {*} outStream - The stream to provide our API over.
    */
   setupControllerConnection(outStream) {
     const api = this.getApi();
-    // the "weak: false" option is for nodejs only (eg unit tests)
-    // it is a workaround for node v12 support
-    const dnode = Dnode(api, { weak: false });
+
     // report new active controller connection
     this.activeControllerConnections += 1;
     this.emit('controllerConnectionChanged', this.activeControllerConnections);
-    // connect dnode api to remote connection
-    pump(outStream, dnode, outStream, (err) => {
-      // report new active controller connection
-      this.activeControllerConnections -= 1;
```

Comment:
It seems like we're not tracking when connections are closed anymore. We keep track of this as part of "quiet mode" - where we stop polling the network when no UI is open.

Reply:
It looks like this plus the `controllerConnectionChanged` event could be emitted in the `end` handler for the stream, to restore this functionality.

Reply:
nice find 🙏